Design/Rule0006PrimaryKeyNotAutoIncrementInTemporaryTable.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Microsoft.Dynamics.Nav.CodeAnalysis;
+using Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics;
+using Microsoft.Dynamics.Nav.CodeAnalysis.Syntax;
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+
+namespace BusinessCentral.LinterCop.Design
+{
+    [DiagnosticAnalyzer]
+    public class Rule0006PrimaryKeyNotAutoIncrementInTemporaryTable : DiagnosticAnalyzer
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create<DiagnosticDescriptor>(DiagnosticDescriptors.Rule0006PrimaryKeyNotAutoIncrementInTemporaryTable);
+
+        public override void Initialize(AnalysisContext context)
+            => context.RegisterSymbolAction(new Action<SymbolAnalysisContext>(this.CheckTablePrimaryKeyIsNotAutoIncrement), SymbolKind.Table);
+
+        private void CheckTablePrimaryKeyIsNotAutoIncrement(SymbolAnalysisContext context)
+        {
+            ITableTypeSymbol tableTypeSymbol = (ITableTypeSymbol)context.Symbol;
+            if (!IsSymbolAccessible(tableTypeSymbol))
+                return;
+
+            CheckTable(tableTypeSymbol, ref context);
+        }
+
+        private void CheckTable(ITableTypeSymbol table, ref SymbolAnalysisContext context) {
+            if (table.TableType != TableTypeKind.Temporary)
+                return;
+
+            foreach (var field in table.PrimaryKey.Fields) {
+                IPropertySymbol propertySymbol = field.GetProperty(PropertyKind.AutoIncrement);
+                if (propertySymbol != null) {
```

Comment:
Also check the value to be true 😁

Reply:
Resolved in df1d4cedce602ed4b5e1834ee2b5c066aee5de1d :)